src/transformations.jl

```diff
@@ -97,22 +99,70 @@ julia> StatsBase.transform(dt, X)
  -1.0   0.0  1.0
 ```
 """
-function fit(::Type{ZScoreTransform}, X::AbstractArray{<:Real,2}; center::Bool=true, scale::Bool=true)
-    d, n = size(X)
-    n >= 2 || error("X must contain at least two columns.")
-
+function fit(::Type{ZScoreTransform}, X::AbstractMatrix{<:Real};
+             dims::Integer=1, center::Bool=true, scale::Bool=true)
+    d, m, s = _fit(ZScoreTransform, X, Val(dims), center, scale)
     T = eltype(X)
-    m, s = mean_and_std(X, 2)
-
     return ZScoreTransform(d, (center ? vec(m) : zeros(T, 0)),
                               (scale ? vec(s) : zeros(T, 0)))
 end
 
-function transform!(y::AbstractVecOrMat{<:Real}, t::ZScoreTransform, x::AbstractVecOrMat{<:Real})
+function fit(::Type{ZScoreTransform}, X::AbstractVector{<:Real}; center::Bool=true, scale::Bool=true)
+    T = eltype(X)
+    m, s = mean_and_std(X)
+    return ZScoreTransform(1, (center ? [m] : zeros(T, 0)),
+                              (scale ? [s] : zeros(T, 0)))
+end
+
+function _fit(::Type{ZScoreTransform}, X::AbstractMatrix{<:Real}, ::Val{1}, center::Bool, scale::Bool)
+    n, d = size(X)
+    n >= 2 || error("X must contain at least two rows.")
+    m, s = mean_and_std(X, 1)
+    return d, m, s
+end
+
+function _fit(::Type{ZScoreTransform}, X::AbstractMatrix{<:Real}, ::Val{2}, center::Bool, scale::Bool)
+    d, n = size(X)
+    n >= 2 || error("X must contain at least two columns.")
+    m, s = mean_and_std(X, 2)
+    return d, m, s
+end
+
+function transform!(y::AbstractMatrix{<:Real}, t::ZScoreTransform, x::AbstractMatrix{<:Real}, ::Val{1})
     d = t.dim
-    size(x,1) == size(y,1) == d || throw(DimensionMismatch("Inconsistent dimensions."))
-    n = size(y,2)
-    size(x,2) == n || throw(DimensionMismatch("Inconsistent dimensions."))
+    size(x,2) == size(y,2) == d || throw(DimensionMismatch("Inconsistent dimensions."))
+    n = size(y,1)
+    size(x,1) == n || throw(DimensionMismatch("Inconsistent dimensions."))
+
+    m = t.mean
+    s = t.scale
+
+    if isempty(m)
```

Comment:
Instead of duplicating the code, just call `transform!` on the transpose. Same below.

Reply:
OK